sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/models/ClientEndpointType.java

```diff
@@ -0,0 +1,39 @@
+package com.azure.messaging.webpubsub.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+/**
+ * Defines values for ClientEndpointType
+ */
+public enum ClientEndpointType {
+    /** Default WebPubSub Client Endpoint. E.g: <code>wss://exampleHost.com/client/hubs/exampleHub</code> */
+    DEFAULT("default"),
+
+    /** MQTT Client Endpoint. E.g: <code>wss://exampleHost.com/client/mqtt/hubs/exampleHub</code> */
+    MQTT("mqtt");
+
+    /** The actual serialized value for a ClientEndpointType. */
+    private final String value;
+
+    ClientEndpointType(String value) { this.value = value; }
+
+    /**
+     * Parses a serialized value to a ClientEndpointType instance.
+     *
+     * @param value the serialized value to parse.
+     * @return the parsed ClientEndpointType object, or null if unable to parse.
+     */
+    @JsonCreator
+    public static ClientEndpointType fromString(String value) {
```

Comment:
Why do we need serialization method here?

Reply:
We don't currently need it. But it seems that most enum classes in the repos define their own serialize and de-serialize methods, so this is just to follow conventions